Flaky-test: ResourceGroupConfigListenerTest.testResourceGroupAttachToNamespace

<!--- 

Instructions for reporting a flaky test using this issue template:

1. Replace [test class] in title and body with the test class name
2. Replace [test method] in title and body with the test method that failed. Multiple methods are flaky, remove the content that refers to the test method.
3. Replace "url here" with a url to an example failure. In the Github Actions workflow run logs, you can right click on the line number to copy a link to the line. Example of such url is https://github.com/apache/pulsar/pull/8892/checks?check_run_id=1531075794#step:9:377 . The logs are available for a limited amount of time (usually for a few weeks).
4. Replace "relevant parts of the exception stacktrace here" with the a few lines of the stack trace that shows at leat the exception message and the line of test code where the stacktrace occured.
5. Replace "full exception stacktrace here" with the full exception stacktrace from logs. This section will be hidded by default.
6. Remove all unused fields / content to unclutter the reported issue. Remove this comment too.

-->
ResourceGroupConfigListenerTest is flaky. The testResourceGroupAttachToNamespace test method fails sporadically.

[example failure](https://github.com/apache/pulsar/pull/11026/checks?check_run_id=2894464546#step:10:571)

```
Error:  Tests run: 14, Failures: 1, Errors: 0, Skipped: 13, Time elapsed: 8.365 s <<< FAILURE! - in org.apache.pulsar.broker.resourcegroup.ResourceGroupConfigListenerTest
Error:  testResourceGroupAttachToNamespace(org.apache.pulsar.broker.resourcegroup.ResourceGroupConfigListenerTest)  Time elapsed: 1.058 s  <<< FAILURE!
org.awaitility.core.ConditionTimeoutException: Assertion condition defined as a lambda expression in org.apache.pulsar.broker.resourcegroup.ResourceGroupConfigListenerTest expected object to not be null within 1 seconds.
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:679)
	at org.apache.pulsar.broker.resourcegroup.ResourceGroupConfigListenerTest.createResourceGroup(ResourceGroupConfigListenerTest.java:67)
	at org.apache.pulsar.broker.resourcegroup.ResourceGroupConfigListenerTest.testResourceGroupAttachToNamespace(ResourceGroupConfigListenerTest.java:122)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.AssertionError: expected object to not be null
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.assertNotNull(Assert.java:942)
	at org.testng.Assert.assertNotNull(Assert.java:926)
	at org.apache.pulsar.broker.resourcegroup.ResourceGroupConfigListenerTest.lambda$createResourceGroup$0(ResourceGroupConfigListenerTest.java:70)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	... 4 more
```

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupConfigListenerTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupConfigListenerTest.java
@@ -22,22 +22,17 @@ import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertNotNull;
 import static org.testng.Assert.assertNull;
 import static org.testng.Assert.assertThrows;
-
 import com.google.common.collect.Sets;
+import java.util.Random;
 import org.apache.pulsar.broker.auth.MockedPulsarServiceBaseTest;
 import org.apache.pulsar.client.admin.PulsarAdminException;
 import org.apache.pulsar.common.policies.data.ClusterData;
-import org.apache.pulsar.common.policies.data.ClusterDataImpl;
 import org.apache.pulsar.common.policies.data.ResourceGroup;
 import org.apache.pulsar.common.policies.data.TenantInfoImpl;
 import org.awaitility.Awaitility;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
-import org.testng.annotations.Ignore;
 import org.testng.annotations.Test;
-
-import java.util.Random;
-import java.util.concurrent.TimeUnit;
 
 public class ResourceGroupConfigListenerTest extends MockedPulsarServiceBaseTest {
 
@@ -64,7 +59,7 @@ public class ResourceGroupConfigListenerTest extends MockedPulsarServiceBaseTest
     public void createResourceGroup(String rgName, ResourceGroup rg) throws PulsarAdminException {
         admin.resourcegroups().createResourceGroup(rgName, rg);
 
-        Awaitility.await().atMost(1, TimeUnit.SECONDS).untilAsserted(() -> {
+        Awaitility.await().untilAsserted(() -> {
             final org.apache.pulsar.broker.resourcegroup.ResourceGroup resourceGroup = pulsar
                     .getResourceGroupServiceManager().resourceGroupGet(rgName);
             assertNotNull(resourceGroup);
@@ -75,14 +70,14 @@ public class ResourceGroupConfigListenerTest extends MockedPulsarServiceBaseTest
 
     public void deleteResourceGroup(String rgName) throws PulsarAdminException {
         admin.resourcegroups().deleteResourceGroup(rgName);
-        Awaitility.await().atMost(1, TimeUnit.SECONDS)
+        Awaitility.await()
                 .untilAsserted(() -> assertNull(pulsar.getResourceGroupServiceManager().resourceGroupGet(rgName)));
     }
 
     public void updateResourceGroup(String rgName, ResourceGroup rg) throws PulsarAdminException {
         testAddRg.setPublishRateInMsgs(200000);
         admin.resourcegroups().updateResourceGroup(rgName, rg);
-        Awaitility.await().atMost(1, TimeUnit.SECONDS).untilAsserted(() -> {
+        Awaitility.await().untilAsserted(() -> {
             final org.apache.pulsar.broker.resourcegroup.ResourceGroup resourceGroup = pulsar
                     .getResourceGroupServiceManager().resourceGroupGet(rgName);
             assertNotNull(resourceGroup);
@@ -126,7 +121,7 @@ public class ResourceGroupConfigListenerTest extends MockedPulsarServiceBaseTest
         admin.namespaces().createNamespace(namespaceName);
         admin.namespaces().setNamespaceResourceGroup(namespaceName, rgName);
 
-        Awaitility.await().atMost(5, TimeUnit.SECONDS).untilAsserted(() ->
+        Awaitility.await().untilAsserted(() ->
                         assertNotNull(pulsar
                                 .getResourceGroupServiceManager()
                                 .getNamespaceResourceGroup(namespaceName)));
@@ -150,20 +145,25 @@ public class ResourceGroupConfigListenerTest extends MockedPulsarServiceBaseTest
             admin.resourcegroups().createResourceGroup(rgName, testAddRg);
         }
 
-        Thread.sleep(1000);
+        Awaitility.await().untilAsserted(() -> {
+            for (int i = 0; i < MAX_RGS; i++) {
+                String rgName = String.format("testRg-%d", i);
+                assertNotNull(pulsar.getResourceGroupServiceManager().resourceGroupGet(rgName));
+            }
+        });
+
 
         for (int i = 0; i < MAX_RGS; i++) {
             String rgName = String.format("testRg-%d", i);
-            assertNotNull(pulsar.getResourceGroupServiceManager().resourceGroupGet(rgName));
             admin.resourcegroups().deleteResourceGroup(rgName);
         }
 
-        Thread.sleep(1000);
-
-        for (int i = 0; i < MAX_RGS; i++) {
-            String rgName = String.format("testRg-%d", i);
-            assertNull(pulsar.getResourceGroupServiceManager().resourceGroupGet(rgName));
-        }
+        Awaitility.await().untilAsserted(() -> {
+            for (int i = 0; i < MAX_RGS; i++) {
+                String rgName = String.format("testRg-%d", i);
+                assertNull(pulsar.getResourceGroupServiceManager().resourceGroupGet(rgName));
+            }
+        });
     }
 
     private void prepareData() throws PulsarAdminException {

